### ISSUE TITLE
Black-oil Polymer Module Compares Iterators into Different Containers

### ISSUE THREAD

**Issue by bska** [OPM/ewoms#496]
The member function [`BlackoilPolymerModule::getSkprwatTable()`](https://github.com/OPM/ewoms/blob/b7efb9c775963347017e25241fb7afd0fd7f4079/ewoms/models/blackoil/blackoilpolymermodules.hh#L478) contains the following [test](https://github.com/OPM/ewoms/blob/b7efb9c775963347017e25241fb7afd0fd7f4079/ewoms/models/blackoil/blackoilpolymermodules.hh#L480-L481)
```C++
const auto iterTable = skprwatTables_.find(tableNumber);
if (iterTable != plymwinjTables_.end()) {
```
Here the `iterTable` iterator is compared to the `end()` of a container different from the one used to create it.  That's undefined behaviour.

**Comment by GitPaean** [comment]
Thanks for spotting it. I will create a fix of it.

**Comment by GitPaean** [comment]
closed by #497.

**Comment by bska** [comment]
> Thanks for spotting it.

I didn't.  I just ran OPM-Simulator's [compareECLFiles_flow+2D_POLYMER_INJECTIVITY regression test](https://github.com/OPM/opm-simulators/blob/076f18c5d5a570bea59d14c407cffe90ccbcc04c/compareECLFiles.cmake#L248-L252) using libstdc++'s debug mode as part of some other work.
